overlord/snapstate/snapstate.go

```diff
@@ -162,6 +158,19 @@ func doInstall(s *state.State, snapst *SnapState, ss *SnapSetup) (*state.TaskSet
 			prev = tasks[len(tasks)-1]
 		}
 
+		// make sure we're not scheduling the removal of the target
+		// revision in the case where the target revision is already in
+		// the sequence.
+		for i := 0; i < currentIndex; i++ {
+			si := seq[i]
+			if si.Revision == ss.Revision() {
+				// we do *not* want to removeInactiveRevision of this one
+				copy(seq[i:], seq[i+1:])
```

Comment:
I think I get used to the weird way of writing `delete(seq, i)` - not sure if that is a good sign though.